compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUTensorCoreVectorization.cpp

```diff
@@ -119,10 +120,132 @@ static Optional<SmallVector<int64_t>> getGPUTCNativeVectorSize(Operation *op) {
   return std::nullopt;
 }
 
+static int getVectorContractOpOperandId(
+  vector::ContractionOp contractOp, Operation *op) {
+  if (contractOp.getLhs() == op->getResult(0)) return 0;
+  if (contractOp.getRhs() == op->getResult(0)) return 1;
+  if (contractOp.getAcc() == op->getResult(0)) return 2;
+  return -1;
+}
+
+// Returns operand id for vector::ContractOp where the vector::TransferReadOp
+// is consumed. 
+static int getVectorContractOpOperandIdForVectorReadOp(Operation *op) {
+  vector::ContractionOp contractOp;
+
+  Operation *firstLevelUser = *((op->getUsers()).begin());
+  if (auto contractOp = dyn_cast<vector::ContractionOp>(firstLevelUser))
+      return getVectorContractOpOperandId(contractOp, op);
+  Operation *secondLevelUser = *((firstLevelUser->getUsers()).begin());
+  if (auto contractOp = dyn_cast<vector::ContractionOp>(secondLevelUser))
+      return getVectorContractOpOperandId(contractOp, firstLevelUser);
+  return -1;
+}
+
+// Helper function to return native size for MMA.SYNC-based operations.
+static Optional<SmallVector<int64_t>> getMmaNativeVectorSize(
+    Operation *op) {
+  
+  // Shape of native Tensor Core GPU mma.sync operations.
+  int64_t mmaShapeM = 16;
+  int64_t mmaShapeN = 8;
+  int64_t mmaShapeK;
+
+  // Shape the mma.sync warp-level operation.
+  if (auto contract = dyn_cast<vector::ContractionOp>(op)) {
+    auto sourceType = contract.getLhsType().getElementType();
```

Comment:
spell out type here, following this convention:
https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable